### PR TITLE
Refactor nonce max fee setting

### LINKED ISSUE #1303: Refactor setting max fee and nonce in subcommands
https://github.com/foundry-rs/starknet-foundry/pull/1283/files#r1417097371

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1303

## Introduced changes

- Introduced apply optional function and used it to remove duplication

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by drknzz — APPROVED** [review]

**Review by drknzz — DISMISSED** [review]
declare is not subcommand**S**

**Review by Arcticae — APPROVED** [review]
@drknzz seems that the TODO was addressed 🤔

File: crates/sncast/src/starknet_commands/deploy.rs [hunk 1]
```diff
@@ -57,7 +57,8 @@ pub async fn deploy(
     let factory = ContractFactory::new(class_hash, account);
     let deployment = factory.deploy(constructor_calldata.clone(), salt, unique);
 
-    // todo: refactor setting max_fee and nonce
+    // todo: use apply_optional here when `Deployment` in starknet-rs is public
+    //  otherwise we cannot pass the necessary reference to a function
```
Comment (cptartur) [inline crates/sncast/src/starknet_commands/deploy.rs:61]: Should we create a new issue for this?

Comment (THenry14) [reply crates/sncast/src/starknet_commands/deploy.rs:61]: yes

**Review by THenry14 — APPROVED** [review]
lgtm

**Pull request merged**